### ISSUE TITLE
shimgen - remove from ReadyToRun builds

### ISSUE THREAD

**Issue by tapika** [tapika/swupd#8]
https://github.com/tapika/swupd/blob/master/src/chocolatey.resources/tools/shimgen.license.txt

Even this shimgen is by default now in disabled state, but it's still included in `ReadyToRun` build, and potentially in coming new installation package. 

Need to remove `shimgen` from further distribution.

Maybe could be build time controlled.

**Comment by tapika** [comment]
`shimgen.exe` is also published via git repository (which is public), so binary needs to be removed from git as well.

**Comment by pauby** [comment]
There have been several releases since this was raised, but shimgen.exe is still in the repository and being distributed, which is [prohibited by the license agreement](https://github.com/tapika/swupd/blob/master/src/chocolatey.resources/tools/shimgen.license.txt#L14-L15).

**Comment by tapika** [comment]
> There have been several releases since this was raised, but shimgen.exe is still in the repository and being distributed, which is [prohibited by the license agreement](https://github.com/tapika/swupd/blob/master/src/chocolatey.resources/tools/shimgen.license.txt#L14-L15).

There are two cases in here:
1. Making fork from `choco` repository also automatically copies `shimgen.exe` tool.
2. `shimgen.exe` published in [releases](https://github.com/tapika/swupd/releases) also included, but it's by default not enabled. (See [link](https://github.com/tapika/swupd/blob/master/LICENSE#L8) )

For 1 - I have analyzed removing `shimgen.exe` - but it breaks integration tests. Fixing those require some time, which I don't necessarily have. Also with 1 - same issue is applicable to remaining [853 forks](https://github.com/chocolatey/choco/network/members) - so same issue persists on all of them. 

`shimgen.exe` also still exists in main choco repository - see [tools](https://github.com/chocolatey/choco/tree/master/src/chocolatey.resources/tools) - so 1 needs to be also fixed from main `choco` repository.

Should I raise ticket for fixing 1 issue to `choco` ?

As for 2 - `ReadyToRun` `choco.exe` executable does contains `shimgen.exe`, but it's in disabled state by default.
So issue is minor from my perspective.

I will fix this issue when have time.

**Comment by pauby** [comment]
> For 1 - I have analyzed removing shimgen.exe - but it breaks integration tests. 

Those tests can be disabled in the meantime.

> Also with 1 - same issue is applicable to remaining [853 forks](https://github.com/chocolatey/choco/network/members) - so same issue persists on all of them.

There is a difference between a fork with an intention to contribute and a fork making a competing product. You can see that when they have been forked to make a competing product, they are required to remove things they were not authorised to use.

**Comment by tapika** [comment]
> There is a difference between a fork with an intention to contribute and a fork making a competing product.

I've tried to make some pull request, than raised ~ 6 ticket for new changes - but since low or no response - started my own fork. 

I've already proposed several times to walk through my fork changes and then realign work so that we would have one fork instead of two - but my changes were either rejected or neglected.

There is no alternative, but to raise second product meanwhile.

So "competing" product is not intentional direction at least.

**Comment by pauby** [comment]
> So "competing" product is not intentional direction at least.

Like other forks that are competing products, they are required to remove things they were not authorised to use.

**Comment by tapika** [comment]
Fixed. I've removed your shimgen binary distribution, also mention about it's license.

I've created new shimgen instead - and compared to your version there were following upgrades:
- choco's `shimgen` based on .net 4 => Tarmo's `shimgen` based on 4.8 framework
- choco's `shim`s based on .net 4 => Tarmo's `shim`s based on .net core 3.1
- choco's `shimgen` by default enabled => Tarmo's choco `shimgen` by default disabled, but can be enabled if necessary

shimgen by default is eating a lot of performance - for example from integration tests - at least there it should be disabled.
but as a feature maybe I enable my own shimgen later on - for now haven't seeing need for it.

Closing as completed.
